Add Crowdin

Kas-tle opened the pull request:
Closes #9.

Still need to optimize deployment strategy as currently it takes about 20 minutes for 10 locales. I am considering either doing a GH actions matrix to build various locales in parallel, adding caching for web thumbnail generation, or both.

See https://geyser.kastle.dev/. Select "Crowdin" from the locale menu to demo in-context translation.

Kas-tle committed:
Begin testing i18n

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

onebeastchris committed:
Start on re-writing the providers page to provide more useful details to improve the setup guide

Kas-tle committed:
Add crowdin in-context demo for non-mdx

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Roolps committed:
Added MCServerHost to hosting providers (#47)

* Added MCServerHost to hosting providers

* Update providers.ts

* Correct grammar and make instructions more concise

---------

Co-authored-by: Kas-tle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Add crowdin-sync system

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

onebeastchris committed:
more work on this

rtm516 committed:
Rework paper page for setup

rtm516 committed:
Don't try and build paper setup as a full page

onebeastchris committed:
adjust not listed message

Kas-tle committed:
Working sync system for translations

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Only build en locale for PRs

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Modify yarn install

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Do not immute cache

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Try with standard yarn install

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Revert to yarn --immutable

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Remove explicit url/file-loader resolutions

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Only log webpack errors

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

onebeastchris committed:
update mod/proxy pages

mcchampions commented:
Maybe you can use [Crowdin Webhooks](https://support.crowdin.com/webhooks/) to [trigger Github Actions](https://docs.github.com/zh/rest/actions/workflows#create-a-workflow-dispatch-event) to sync translations with Crowdin. This way, you can avoid syncing on every commit.

Kas-tle commented:
The sync time is negligible (under 20 seconds). The major build time issue is the fact that docusuarus must build each locale separately. I could probably shave off 30-50% with caching og images, but I think I could achieve the largest savings by building all languages separately in a matrix, and then combining the builds after the fact. I just need to see if docuusaurus sites built to an individual locale can be successfully combined.

onebeastchris committed:
Update supported Bedrock versions

Tydium committed:
Remove and Replace "TydiumCraft Skin API" (#53)

* Update community-geyser-projects.md

I no longer will run TydiumCraft Skin API. I have no interest in continuing it at this time (or anything minecraft related). However I have switched out the spot with an API that 100% is always updated and active. Hosted by my good friend Bret06. This works so much better as well. Thanks for everything over the years guys! Tydium out!

* Update wiki/other/community-geyser-projects.md

:thumbsup:

Co-authored-by: chris <github@onechris.mozmail.com>

---------

Co-authored-by: chris <github@onechris.mozmail.com>

Creeperz653 committed:
Update wording for creepernation api

Update wording because it’s bothering me

letsgoawaydev committed:
Update GeyserExtras description  (#52)

* Update community-geyser-projects.md

* Update wiki/other/community-geyser-projects.md

Co-authored-by: chris <github@onechris.mozmail.com>

---------

Co-authored-by: Luke <32024335+lukeeey@users.noreply.github.com>
Co-authored-by: chris <github@onechris.mozmail.com>

Kas-tle committed:
Try matrix build strategy

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Fix matrix output

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Matrix goes to GITHUB_OUTPUT

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Fix deploy needs to combine

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Ensure correct commit is used if crowdin updates

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

D4n2021 committed:
Update index.md to make clear that this is for servers, not purely for clients (#57)

santevemox committed:
Update providers.ts (#58)

* Update providers.ts

VemoxHost added

* Maintain alphabetized list

---------

Co-authored-by: Kas-tle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle commented:
Using a build matrix for languages has reduced the build time to about 5.5 minutes:
https://github.com/Kas-tle/GeyserWebsite/actions/runs/13575720560

Kas-tle committed:
Make sure excluded content is respected

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Allow permission for commit action write

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

github-actions[bot] committed:
Sync translations with Crowdin

Kas-tle committed:
Initial setup i18n

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Add ability to hide proxy setup instructions by provider

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Temporary auto deployment; fix broken links

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Fix heading consistency

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

onebeastchris committed:
Update language on provider instructions

onebeastchris committed:
Merge remote-tracking branch 'upstream/master' into fork/Kas-tle/feature/i18n

# Conflicts:
#	src/data/providers.ts

onebeastchris committed:
Update language on new provider instructions

onebeastchris committed:
Documentation for changes in API/2.6.2 (#62)

* Document latest Geyser changes:
- remote resource packs

* Update/Finish pack docs

* small typo fixes

* Update page content

* fix build

* Update supported Bedrock version

Kas-tle committed:
Merge pull request #1 from onebeastchris/feature/i18n

Update host descriptions

triphora committed:
Add Modrinth Servers to providers (#63)

onebeastchris committed:
Update supported versions & api version

ZomkaDEV committed:
Update versions.json to include 1.21.72 (#64)

Novampr committed:
Update bedrock version (#67)

Kas-tle committed:
Add 'we are not mojang' disclaimer to footer

Kas-tle committed:
Stop translations from overriding footer message

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Jens-Co committed:
Fix Standalone dump view (#68)

OrionNodes committed:
Add OrionNodes to providers (#69)

ZomkaDEV committed:
Update supported Bedrock versions in versions.json (1.21.81, 1.21.82) (#70)

Kas-tle committed:
Upgrade actions/setup-node due to hang issue

Kas-tle committed:
Both

Kas-tle committed:
Forgot how many times we setup node...

sylenthuntress committed:
Update faq.mdx to include recommended API (#73)

Tydium skin API is abandoned, please direct users to a working API such as the one recommended by Tydium

Derived from: https://github.com/GeyserMC/GeyserWebsite/commit/19ae50f95ddbfc24ff34a42d278bdc263e0200e6

oryxel1 committed:
Update anticheat-compatibility.md (#74)

onebeastchris committed:
Update versions.json

eclipseisoffline committed:
Update versions.json (#75)

Onako2 committed:
Remove LAC because it is no longer being maintained (#76)

eclipseisoffline committed:
Update versions.json for 1.21.7 (#77)

eclipseisoffline committed:
Update versions.json for 1.21.93 (#78)

eclipseisoffline committed:
Update versions.json for 1.21.94 (#79)

eclipseisoffline committed:
Update versions.json for Java 1.21.8 (#80)

mcchampions committed:
Update Geyser Api version (#81)

Novampr committed:
Readd Hydraulic and add Rainbow, provide Item API V2 warning for Hydraulic (#82)

drakon64 committed:
Fix incorrect Ubuntu on Oracle Cloud instructions (#84)

SleepyPopp committed:
Remove Geyser Android App and Tempest (#83)

SleepyPopp committed:
Update version.json for 1.21.100 (#88)

SleepyPopp committed:
Update getting-started-with-the-api.md (#89)

Novampr committed:
Add and update wiki pages, add Thunder download. (#90)

* Add a Thunder and Rainbow wiki page, touch up some other pages

* Fix mobile sidebar

* Address review 1/2

* Address review 1.5/2 (oops)

* Address review 2/2

Co-authored-by: chris <github@onechris.mozmail.com>

---------

Co-authored-by: chris <github@onechris.mozmail.com>

SleepyPopp committed:
Indicate 1.21.101 on the Geyser Website (#93)

oryxel1 committed:
Added GeyserFloatingPoints to FAQ (#92)

* Added GeyserFloatingPoints.

* Update wiki/geyser/faq.mdx

Co-authored-by: chris <github@onechris.mozmail.com>

---------

Co-authored-by: chris <github@onechris.mozmail.com>

eclipseisoffline committed:
Update versions.json to indicate support for bedrock 1.21.111

eclipseisoffline committed:
Update versions.json for Java 1.21.9/1.21.10 (#97)

* Update versions.json for Java 1.21.9

* Update versions.json to include 1.21.10

* Update supported Bedrock version

---------

Co-authored-by: Aurorawr <auroranova8756@gmail.com>
Co-authored-by: chris <github@onechris.mozmail.com>

Novampr committed:
Update extensions file for new dependency system (#98)

* Update extensions file for new dependency system

* Update version

* Update wiki/geyser/extensions.md

---------

Co-authored-by: chris <github@onechris.mozmail.com>

ZomkaDEV committed:
Replace the Crowdin link with the current one (#101)

* Replace the Crowdin link with the current one

* Replace the link with the subdomain redirect

SleepyPopp committed:
Update versions.json to include 1.21.114 (#102)

SleepyPopp committed:
Update rainbow.md to indicate 1.21.9/.10 (#103)

rtm516 committed:
Revise port forwarding instructions for Ubuntu

Updated instructions for allowing Minecraft and Geyser through the OS firewall on Ubuntu.

Kas-tle committed:
Update providers and deploy action

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

github-actions[bot] committed:
Sync translations with Crowdin

Kas-tle committed:
Merge branch 'master' into feature/i18n

Kas-tle committed:
Correct providers syntax

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

onebeastchris committed:
Use more templates wherever possible

Kas-tle committed:
Add build cache; update docusaurus-og

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Test deploy time with cache hit

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Use read-only cache for matrix build

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Unlikely we will use gh actions on arm...

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
This time cache will hit

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Hopefully correct yarn cache path...

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Maybe npm?

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Surely...

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
This time...

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Kas-tle committed:
Include yarn/docusaurus cache for PRs

Signed-off-by: Joshua Castle <26531652+Kas-tle@users.noreply.github.com>

Novampr committed:
Merge master

Novampr committed:
Add page IDs to more pages, fix broken pages

Novampr committed:
Update i18n pages

github-actions[bot] committed:
Sync translations with Crowdin

Novampr committed:
Add missing support pages and articles on providers, add Discord links
